### DISCUSSION TITLE
Error running python setup.py build command

### DISCUSSION THREAD

**Discussion by wsy-00** [pyswmm/swmm-python#116] [Q&A] [answered]
Hello!

I am attempting to build the swmm-python package by compiling the source code. Firstly, I successfully installed swmm and generated runswmm.exe in the swmm-solver directory. After that, I ran the command "python setup.py build" within a virtual environment. However, I encountered an error as shown in the attached image. Could you please help me understand the reason behind this issue?

Thank you for your assistance.
![7b17ce1897283ea3eafefdf52c5ed10](https://github.com/pyswmm/swmm-python/assets/60472366/bff31568-7fdc-48af-a820-5f11111317b5)
![f0e622c6c94300815548099e15c89c4](https://github.com/pyswmm/swmm-python/assets/60472366/ac259636-7b4d-43cf-ab2b-59480f87ddbc)
![34cecb0864182b4f9f06594f9ce6e96](https://github.com/pyswmm/swmm-python/assets/60472366/b0166f33-4c85-441b-8d7e-67e24b07a152)

**Comment by karosc** [comment]
@wsy-00, what version of cmake are you running?

**Reply by wsy-00** [reply]
I'm using cmake 3.26.4.

**Comment by karosc** [comment, marked as answer]
@wsy-00, I was able to replicate your error and resolved it by downgrading swig to 4.0.2. I think we need to update our interface files and maybe our CMakeLists to make them compatible with a newer version of swig.

Try installing swig 4.0.2 from PyPI and let us know if that fixes it for you!
`pip install swig==4.0.2`

**Reply by wsy-00** [reply]
Thank you very much, it works now!
